src/main/java/com/lpvs/controller/GitHubWebhooksController.java

```diff
@@ -37,17 +44,20 @@ public class GitHubWebhooksController {
 
     private static Logger LOG = LoggerFactory.getLogger(GitHubWebhooksController.class);
 
-    private static final String SIGNATURE = "X-Hub-Signature";
+    private static final String SIGNATURE = "X-Hub-Signature-256";
     private static final String SUCCESS = "Success";
     private static final String ERROR = "Error";
+    private static final String ALGORITHM = "HmacSHA256";
 
     @RequestMapping(value = "/webhooks", method = RequestMethod.POST)
-    public ResponseEntity<ResponseWrapper> gitHubWebhooks(@RequestHeader(SIGNATURE) String signature, @RequestBody String payload) throws InterruptedException {
+    public ResponseEntity<ResponseWrapper> gitHubWebhooks(@RequestHeader(SIGNATURE) String signature, @RequestBody String payload) throws Exception {
         LOG.info("New webhook request received");
 
         // if signature is empty return 401
         if (!StringUtils.hasText(signature)) {
             return new ResponseEntity<>(new ResponseWrapper(ERROR), HttpStatus.FORBIDDEN);
+        } else if (!GITHUB_SECRET.isBlank() && wrongSecret(signature, payload)) {
```

Comment:
May we try to change this line as follows?

```suggestion
        } else if (!GITHUB_SECRET.trim().isEmpty() && wrongSecret(signature, payload)) {
```